src/Playwright/Core/Page.cs

```diff
@@ -337,7 +337,7 @@ public Task<IResponse> RunAndWaitForNavigationAsync(Func<Task> action, PageRunAn
             });
 
         public Task<IRequest> WaitForRequestAsync(string urlOrPredicate, PageWaitForRequestOptions options = default)
-            => InnerWaitForEventAsync(PageEvent.Request, null, e => e.Url.UrlMatches(urlOrPredicate), options?.Timeout);
+            => InnerWaitForEventAsync(PageEvent.Request, null, e => e.Url.UrlMatches(urlOrPredicate.CombineUrlWithBase(Context.BaseUrl)), options?.Timeout);
```

Comment:
Instead of merging it everywhere I would pass the baseURL inside URLMatches like we do in Python and if baseURL is given + does not start with * I would join.

(and below)

Reply:
It's just a preference of using an extension method. I don't think it makes a difference.